src/bot-handle-comment.ts

```diff
@@ -10,7 +10,7 @@ import { tipUser } from "./tip";
 import { ContributorAccount, GithubReactionType, State, TipRequest, TipSize } from "./types";
 import { formatTipSize, getTipSize, parseContributorAccount } from "./util";
 
-type OnIssueCommentResult = { type: "success"; message: string } | { type: "error"; errorMessage: string | undefined };
+type OnIssueCommentResult = { success: true; message: string } | { success: false; errorMessage: string | undefined };
```

Comment:
Wouldn't it be simpler and more js-native just to throw errors, instead of returning `errorMessage`?  
Especially, if you're already using it for `catch` blocks, meaning `errorMessage` can already contain arbitrary error messages

Reply:
I generally agree with that, I also thought about it.

However - there is a difference in how we treat a returned `errorMessage` and a thrown error.

The returned `errorMessage` is controlled by us, we know it's user-friendly (e.g. `Contributor and tipper cannot be the same person`), and we make it public (by commenting on the PR).

A thrown error is unknown, might be gibberish for the user, might be unsafe (who knowns if it wouldn't print out some secret variables). We don't make it public, we log it privately.

Reply:
That was precisely my point here: 

> Especially, if you're already using it for `catch` blocks, meaning `errorMessage` can already contain arbitrary error messages

Reply:
The intention was to NOT have unknown things in `errorMessage`.

Like here, on a catch I omit the unkown `errorMessage` so it won't be made public.

https://github.com/paritytech/substrate-tip-bot/blob/6aa61be3c452aa4702b4b2bb52e350abe63973f7/src/tip.ts#L30

However now I realize it's not 100% like that.

https://github.com/paritytech/substrate-tip-bot/blob/6aa61be3c452aa4702b4b2bb52e350abe63973f7/src/bot-handle-comment.ts#L101-L114

Here something unsafe or gibberish would get public on a github comment.

So I think there are two ways:

- Change those two try-catches to differentiate between known user-friendly error messages and unknown errors
- Get rid of this idea and simply treat every error the same way

Reply:
Or - a third option came to me after some thoughts - I could throw an internal error class such as `UserFriendlyError` or something and based on that make it public or not.

Reply:
> Change those two try-catches to differentiate between known user-friendly error messages and unknown errors

Yeah, in this case, I would just drop the catches, and add a surrounding catch one level above.

Reply:
I've updated the `getTipSize` and `parseContributorAccount` util functions consistent with what I already had in `tipSizeToOpenGovTrack`.

Now there is a full distinction between a known, user-friendly error message and an unknown error thrown.